components/devicecloud-api/org.wso2.carbon.device.mgt.iot.common.api/src/main/java/org/wso2/carbon/device/mgt/iot/common/api/GroupManagerService.java

```diff
@@ -0,0 +1,252 @@
+            List<Group> groups = new GroupManagement().getGroupManagementService().getGroupListOfUser(username);
+            Group[] groupArray = new Group[groups.size()];
+            return groups.toArray(groupArray);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Path("/group/all/count")
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    public int getGroupCountOfUser(@FormParam("username") String username) {
+        try {
+            return new GroupManagement().getGroupManagementService().getGroupCountOfUser(username);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return 0;
+        }
+    }
+
+    @Path("/group/id/{groupId}/share")
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    public boolean shareGroup(@FormParam("username") String username, @FormParam("shareUser") String shareUser, @PathParam("groupId") int groupId, @FormParam("role") String sharingRole) {
+        try {
+            return new GroupManagement().getGroupManagementService().shareGroup(shareUser, groupId, sharingRole);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    @Path("/group/id/{groupId}/unshare")
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    public boolean unShareGroup(@FormParam("username") String username, @FormParam("unShareUser") String unShareUser, @PathParam("groupId") int groupId, @FormParam("role") String sharingRole) {
+        try {
+            return new GroupManagement().getGroupManagementService().unShareGroup(unShareUser, groupId, sharingRole);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    @Path("/group/id/{groupId}/role")
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    public boolean addNewSharingRoleForGroup(@FormParam("username") String username, @PathParam("groupId") int groupId, @FormParam("role") String roleName, @FormParam("permissions") String[] permissions) {
+        try {
+            Permission[] perms = new Permission[permissions.length];
+            for (int i = 0; i < permissions.length; i++) {
+                perms[i] = new Permission("group-mgt/user", permissions[i]);
+            }
+            return new GroupManagement().getGroupManagementService().addNewSharingRoleForGroup(username, groupId, roleName, perms);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    @Path("/group/id/{groupId}/role")
+    @DELETE
+    @Consumes("application/json")
+    @Produces("application/json")
+    public boolean removeSharingRoleForGroup(@PathParam("groupId") int groupId, @FormParam("role") String roleName) {
```

Comment:
Ack, Already done in latest repository. Code moved to https://github.com/wso2/carbon-device-mgt